File: study-tracker/frontend/src/api/ai.js
import client from './client';

export const getNotes = async (subjectId) => {
    const response = await client.get(`/api/ai/subjects/${subjectId}/notes`);
    return response.data;
};

export const generateNote = async (subjectId, promptType) => {
    const response = await client.post(`/api/ai/subjects/${subjectId}/notes`, { prompt_type: promptType });
    return response.data;
};

export const getChatHistory = async (subjectId) => {
    const response = await client.get(`/api/ai/subjects/${subjectId}/chat`);
    return response.data;
};

export const sendChatMessage = async (subjectId, message) => {
    const response = await client.post(`/api/ai/subjects/${subjectId}/chat`, { message });
    return response.data;
};

File: study-tracker/backend/app/routes/ai.py
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session
from app.core.database import get_db
from app.models.schemas import AINoteOut, AIChatMessageOut, AIChatRequest, AINoteGenerateRequest
from app.services import ai_service, subject_service
from app.routes.auth import get_current_user_id

router = APIRouter()

@router.get("/subjects/{subject_id}/notes", response_model=list[AINoteOut])
def get_notes(subject_id: int, user_id: int = Depends(get_current_user_id), db: Session = Depends(get_db)):
    subject_service.verify_subject_owner(db, subject_id, user_id)
    return ai_service.get_notes(db, subject_id)

@router.post("/subjects/{subject_id}/notes", response_model=AINoteOut)
def generate_note(subject_id: int, payload: AINoteGenerateRequest, user_id: int = Depends(get_current_user_id), db: Session = Depends(get_db)):
    subject_service.verify_subject_owner(db, subject_id, user_id)
    return ai_service.generate_notes(db, subject_id, payload.prompt_type)

@router.get("/subjects/{subject_id}/chat", response_model=list[AIChatMessageOut])
def get_chat(subject_id: int, user_id: int = Depends(get_current_user_id), db: Session = Depends(get_db)):
    subject_service.verify_subject_owner(db, subject_id, user_id)
    return ai_service.get_chat_history(db, subject_id)

@router.post("/subjects/{subject_id}/chat", response_model=AIChatMessageOut)
def send_chat(subject_id: int, payload: AIChatRequest, user_id: int = Depends(get_current_user_id), db: Session = Depends(get_db)):
    subject_service.verify_subject_owner(db, subject_id, user_id)
    return ai_service.send_chat_message(db, subject_id, payload.message)
